feat: add a depends-on conditions (#2619)
Mergifyio/mergify-engine: docs/source/actions/queue.rst

Merging broken pull requests can happen when outdated pull requests are being
merged in their base branch.

Mergify always respects the `branch protection`_ settings. When the conditions
match and the ``queue`` action runs, Mergify waits for the branch protection to
be validated before embarking and merging the pull request.

.. _`branch protection`: https://docs.github.com/en/github/administering-a-repository/about-protected-branches

> It can also wait for other pull request to get merged first (see
> :ref:`depends-on`).

Reviewer: ```suggestion
Mergify also waits for dependent pull requests to get merged first (see :ref:`depends-on`).
```

          queue:
            name: default

.. _merge-depends-on:

⛓️ Defining Pull Request Dependencies
-------------------------------------

|premium plan tag|
|open source plan tag|

You can specify dependencies between pull requests from the same repository.
Mergify waits for the linked pull requests to be merged before merging any pull

> request with a ``Depends-on:`` header.

Reviewer: ```suggestion
request with a ``Depends-On:`` header.
```

          queue:
            name: default

.. _depends-on:

⛓️ Defining the pull request dependencies
----------------------------------------

|premium plan tag|

Mergify can wait for other pull requests to be merged before embarking this one.

To do this you must add a ``Depends-On:`` lines to you pull request body:

```
New awesome feature 🎉

To get the full picture, you may need to look at these pull requests:

Depends-On: #42
Depends-On: #123

> ```

Reviewer: same as above
